### ISSUE TITLE
Dev Containers extension unable to attach to running podman container

### ISSUE THREAD

**Issue by muralidharanram** [microsoft/vscode-remote-release#7970]
- VSCode Version: 1.75.0 & 1.76.0-insiders
- Local OS Version: Fedora 37 
- Remote OS Version: redhat/ubi9:latest
- Remote Extension/Connection Type: Containers

podman version : 4.3.1
podman compose version : 1.0.3

Followed instructions from https://code.visualstudio.com/remote/advancedcontainers/docker-options#_podman to update the settings as shown in the step#2 below

- Logs:
```
[422 ms] Dev Containers 0.275.0 in VS Code 1.75.0 (e2816fe719a4026ffa1ee0189dc89bdfdbafb164).
[421 ms] Start: Resolving Remote
[423 ms] Setting up container: dev01
[428 ms] Start: Run: podman inspect --type container dev01
[463 ms] Start: Check Docker is running
[463 ms] Start: Run: podman version --format {{.Server.APIVersion}}
[489 ms] Server API version: 4.3.1
[490 ms] Start: Run: /usr/share/code/code --ms-enable-electron-run-as-node /home/user/.vscode/extensions/ms-vscode-remote.remote-containers-0.275.0/dist/spec-node/devContainersSpecCLI.js set-up --container-id cd3e0ba22041c3df47ae41e18e9727d8e9bf24486422320f3e1a88a38f47b687 --user-data-folder /home/user/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --docker-path podman --docker-compose-path podman-compose --container-session-data-folder tmp/devcontainers-258455ed-9bbe-4cd2-bacb-4689eb6994181675499892816 --log-level debug --log-format json --config /home/user/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/imageConfigs/docker.io%2fmystarkfuture%2faio-dev%3alatest.json --default-user-env-probe loginInteractiveShell --skip-post-create
[698 ms] devcontainer set-up
[698 ms] Set up an existing container as a dev container
[698 ms] Options:
[698 ms]   --help                           Show help  [boolean]
[699 ms]   --version                        Show version number  [boolean]
[699 ms]   --docker-path                    Docker CLI path.  [string]
[699 ms]   --container-data-folder          Container data folder where user data inside the container will be stored.  [string]
[699 ms]   --container-system-data-folder   Container system data folder where system data inside the container will be stored.  [string]
[699 ms]   --container-id                   Id of the container.  [string] [required]
[699 ms]   --config                         devcontainer.json path.  [string]
[699 ms]   --log-level                      Log level for the --terminal-log-file. When set to trace, the log level for --log-file will also be set to trace.  [choices: "info", "debug", "trace"] [default: "info"]
[699 ms]   --log-format                     Log format.  [choices: "text", "json"] [default: "text"]
[699 ms]   --terminal-columns               Number of rows to render the output for. This is required for some of the subprocesses to correctly render their output.  [number]
[700 ms]   --terminal-rows                  Number of columns to render the output for. This is required for some of the subprocesses to correctly render their output.  [number]
[700 ms]   --default-user-env-probe         Default value for the devcontainer.json's "userEnvProbe".  [choices: "none", "loginInteractiveShell", "interactiveShell", "loginShell"] [default: "loginInteractiveShell"]
[700 ms]   --skip-post-create               Do not run onCreateCommand, updateContentCommand, postCreateCommand, postStartCommand or postAttachCommand and do not install dotfiles.  [boolean] [default: false]
[700 ms]   --skip-non-blocking-commands     Stop running user commands after running the command configured with waitFor or the updateContentCommand by default.  [boolean] [default: false]
[700 ms]   --user-data-folder               Host path to a directory that is intended to be persisted and share state between sessions.  [string]
[700 ms]   --remote-env                     Remote environment variables of the format name=value. These will be added when executing the user commands.  [string]
[700 ms]   --dotfiles-repository            URL of a dotfiles Git repository (e.g., https://github.com/owner/repository.git)  [string]
[700 ms]   --dotfiles-install-command       The command to run after cloning the dotfiles repository. Defaults to run the first file of `install.sh`, `install`, `bootstrap.sh`, `bootstrap`, `setup.sh` and `setup` found in the dotfiles repository`s root folder.  [string]
[700 ms]   --dotfiles-target-path           The path to clone the dotfiles repository to. Defaults to `~/dotfiles`.  [string] [default: "~/dotfiles"]
[700 ms]   --container-session-data-folder  Folder to cache CLI data, for example userEnvProb results  [string]
[700 ms] Unknown arguments: docker-compose-path, dockerComposePath
[702 ms] Exit code 1
[707 ms] Command failed: /usr/share/code/code --ms-enable-electron-run-as-node /home/user/.vscode/extensions/ms-vscode-remote.remote-containers-0.275.0/dist/spec-node/devContainersSpecCLI.js set-up --container-id cd3e0ba22041c3df47ae41e18e9727d8e9bf24486422320f3e1a88a38f47b687 --user-data-folder /home/user/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --docker-path podman --docker-compose-path podman-compose --container-session-data-folder tmp/devcontainers-258455ed-9bbe-4cd2-bacb-4689eb6994181675499892816 --log-level debug --log-format json --config /home/user/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/imageConfigs/docker.io%2fmystarkfuture%2faio-dev%3alatest.json --default-user-env-probe loginInteractiveShell --skip-post-create
[707 ms] Exit code 1
```

Steps to Reproduce:
1. start a container outside of vscode `podman run --name ubi1 -d -t docker.io/redhat/ubi9:latest`
2. In vscode change the Dev Container settings as  defined in https://code.visualstudio.com/remote/advancedcontainers/docker-options#_podman.
Docker Path : podman
Docker Compose Path : podman-compose
settings.json as shown below
```
{
    "workbench.colorTheme": "Default Dark+",
    "window.titleBarStyle": "custom",
    "dev.containers.dockerComposePath": "podman-compose",
    "dev.containers.dockerPath": "podman",
    "dev.containers.defaultFeatures": {

        "runArgs": [
            "--userns=keep-id"
        ],
        "containerEnv": {
            "HOME": "/home/node"
        }
    }
}
```
3. Restart vscode. 
4. In the Containers panel, select `Attach to Container` to attach vscode to the container we started in step#1. Output tab for Dev Containers outputs the log above and it fails to attach.

using `Extensions: install a specific version of Extension...` downgrading to the earlier version of the extension (v0.266.1) resolves the issue. VS code is able to attach to the running podman container.

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

**Comment by chrmarti** [comment]
Steps to reproduce / verify:

- Change the `dev.containers.dockerComposePath` user setting to a non-default value (executable does not have to exist).
- Trying to attach to an existing container will fail.

Verify by ensuring attaching works.

(Make sure to revert the user setting in the end.)

**Comment by chrmarti** [comment]
Fixed in Dev Containers 0.275.1 and 0.277.0-pre-release.
